signature/sign.go

```diff
@@ -113,16 +120,28 @@ func Sign(_ context.Context, key jwk.Key, sf SignedFielder, opts ...Option) (*pi
 		return nil, err
 	}
 
-	if options.logger != nil {
+	switch key := key.(type) {
```

Comment:
This is new to me!
I haven't done serious Golang development for a while, you used to need to do Reflect here but that was pretty slow. Is this some newish Golang feature?

Reply:
Type switches have been around for a very long time: https://web.archive.org/web/20120331010210/golang.org/ref/spec

Reply:
How had I never come across this before!
I needed the magic words to google, it's called a TypeSwitchGuard